### PR TITLE
Tutorial game logic

### PR DESCRIPTION
**Pull request by Marty99-dev**

### CODE REVIEW

**Review by JorisMonnet — CHANGES_REQUESTED** [review]
Read my review on the old PR please, same problems

Log
Bin files
...

**Comment by Marty99-dev** [comment]
> Read my review on the old PR please, same problems
> 
> Log Bin files ...

Good comment ! I'm doing this at the moment.

File: app/src/main/java/ch/epfl/sdp/blindwar/tutorial/DemoActivity.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    private lateinit var scoreTextView: TextView
+    private lateinit var musicMetaData: MusicMetaData
+    private lateinit var visualizer: CircleVisualizer
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_demo)
+
+        // Create the game instance with difficulty easy
+        this.game = GameTutorial(assets, 5000)
```
Comment (marluxiaboss) [inline app/src/main/java/ch/epfl/sdp/blindwar/tutorial/DemoActivity.kt:25]: It should be more clear that 5000 corresponds to the difficulty easy. Difficulties should be better managed/more clear in the code if it's supposed to be a real feature.

**Review by marluxiaboss — APPROVED** [review]
The tests are using hamcrest which make them clear. There are some parts where the values(int, string as well) could be constants.

File: app/src/main/java/ch/epfl/sdp/blindwar/tutorial/DemoActivity.kt (hunk 1 above)
Comment (Marty99-dev) [reply app/src/main/java/ch/epfl/sdp/blindwar/tutorial/DemoActivity.kt:25]: I have planned to  do that while working on the solo mode ! But good remark !

**Review by ghemDD — APPROVED** [review]
We still have to correct the small bug signaled in the old PR in GameTutorial in nextRound() but apart from that it should be good 
( demoActivity provokes an Exception when all sample music from the tutorial have already been played in method nextRound (TODO: add check for emptiness of playlist in the beginning of the method))

**Review by JorisMonnet — APPROVED** [review]

**Pull request merged**